design/cstor/v1/cstorpool.md

```diff
@@ -245,11 +245,24 @@ type CStorPoolInstanceStatus struct {
 	ReadOnly bool `json:"readOnly"`
 }
 
-// CStorPoolInstanceCapacity stores the pool capacity related attributes.
+// CStorPoolCapacityAttr stores the pool capacity related attributes.
 type CStorPoolInstanceCapacity struct {
+	// The amount of storage space within the pool that has been
+	// physically allocated on disk
+	Used resource.Quantity `json:"used"`
```

Comment:
Can we rename this to `PhysicalUsed` to make it easy to compare between LogicalUsed and PhyscialUsed?

Reply:
On second thoughts, let us assume we retain the existing attributes to mean: 
Total => Total Capacity of the pool excluding parity
Used => Amount of physical data (and its metadata) written to pool after applying compression, etc. 
Free => is ideally Total - Used. 

Now we need attributes to store additional information, which are possibly ZFS specific like:
- "LogicalUsed" which is about total data + metadata blocks written without applying compressions
- "TotalPhysical" which is the sum of the capacity of all the disks from raid/data groups. 

Just to avoid confusion between the Total, Used and Free with ZFS specific attributes, can there be a place holder for these advanced parameters either as a sub-type under CStorPoolInstanceCapacity or maybe another block that just stores extra ZFS properties.

Reply:
That make sense... I will add those advanced parameters as other fields in CStorPoolInstanceCapacity.

Reply:
Updated the PR as suggested PTAL

Reply:
Yes @kmova , makes sense.

Reply:
@mittachaitu, @kmova , will it be fine if we state an example here of raidz1 and/or mirror pool . The correlation between ZFS and CustomResource will become more evident.

Reply:
@avishnu Updated the required details in the PR description.

Reply:
n_P :
Tried to become a little specific. How about the following comment?
```
// Amount of physical data (and its metadata) written to pool
after applying compression, excluding storage used to store parity, etc.
```